department-of-veterans-affairs/abd-vro
# Clarify BIP API Resources Directory

In our ongoing effort to stabilize the unit and integration tests, I've consolidated the resources directory of the svc-bip-api. I've also modified how the autowiring of restTemplate instances was happening.

## What was the problem?
Still seeing inconsistent and difficult to understand configuration issues in the svc-bip-api project.

Associated tickets or Slack threads:
- [#2216](https://github.com/department-of-veterans-affairs/abd-vro/issues/2216)

## How does this fix it?[^1]
Eliminates the duplicate resources directories and clarifies how spring should construct the resttemplate class.

## How to test this PR
- To run integration tests, run `./scripts/bip-service.sh`
- To run unit tests, `./gradlew :svc-bip-api:build --tests '*BipApiServiceTest*'`


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
